rule_schema_v1.yaml

```diff
@@ -6,17 +6,103 @@ $schema: http://json-schema.org/draft-07/schema#
 #!!If you add new syntax to this file, you probably need to add some EXPERIMENTAL
 #  comment before!!
 $defs:
+  new-pattern:
+    title: "Return finding where code matches against the following pattern"
+    oneOf:
+    - type: string
+    - type: object
+      oneOf:
+        - required: [ pattern ]
+        - required: [ regex ]
+        - required: [ and ]
```

Comment:
I thought there was consensus in renaming `and` as `all` and `or` as `any`.

Reply:
was there consensus? i recall a discussion, but i dont feel qualified to make the call either way at this stage

i would be inclined to ship it with this syntax for now, and then test this with SR over the course of the next few weeks (i think most of the people weighing in were not security researchers). is that OK?

Reply:
I recall bence had a pretty strong argument.
Note that security researchers do not have the final words on semgrep syntax. While our main target right now are AppSec people for semgrep, our long-term goal for semgrep is that every developers use Semgrep to enforce coding rules, finding bugs, enforcing security guardrails, etc. Therefore we should aspire to not only please security researchers but more generally every developers (hard to do, I know).

Reply:
what i mean is that they have the most experience writing rules. i would be inclined to test it both ways, but they are professional users of the tool, so I would be inclined to see what they think

Reply:
anyways, ill make the change, but i am interested in what users think